tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/Generators/ReadMeGeneratorTool.cs

```diff
@@ -0,0 +1,219 @@
+using OpenAI.Chat;
+
+namespace Azure.Sdk.Tools.Cli.Tools.Generators
+{
+    // Will add after we decide which tools are exported via the MCP.
+    //[McpServerToolType, Description("Generates a README file, using service documentation")]
+    public partial class ReadMeGeneratorTool : MCPTool
+    {
+        private readonly ILogger<ReadMeGeneratorTool> logger;
+        private readonly IOutputService output;
+        private readonly AzureOpenAIClient openAiClient;
+        private Option<string> packagePathOption;
+        private Option<string> engPathOption;
+        private Option<string> outputPathOption;
+        private Option<string> templatePathOption;
+        private Option<string> serviceDocumentationOption;
+        private Option<string> modelOption;
+
+        public ReadMeGeneratorTool(ILogger<ReadMeGeneratorTool> logger, IOutputService output, AzureOpenAIClient openAiClient)
+        {
+            ArgumentNullException.ThrowIfNull(logger, nameof(logger));
+            ArgumentNullException.ThrowIfNull(output, nameof(output));
+
+            // This should be removed, as it won't be possible to have `null` here when the
+            // patterns for DI are fixed.
+            ArgumentNullException.ThrowIfNull(openAiClient, nameof(openAiClient));
+            this.logger = logger;
+            this.output = output;
+            this.openAiClient = openAiClient;
+
+            this.CommandHierarchy = [SharedCommandGroups.Generators];
+        }
+
+        public override Command GetCommand()
+        {
+            templatePathOption = new Option<string>("--template-path")
+            {
+                IsRequired = true,
+                Description = "Path to the README template file (ie: Templates/README-template.go.md)",
+            };
+
+            templatePathOption.SetDefaultValue("Templates/ReadMeGenerator/README-template.go.md");
+
+            serviceDocumentationOption = new Option<string>(
+                "--service-documentation")
+            {
+                Description = "URL to the service documentation",
+                IsRequired = true
+            };
+
+            serviceDocumentationOption.SetDefaultValue("https://learn.microsoft.com/en-us/azure/service-bus-messaging/");
+
+            packagePathOption = new Option<string>(
+                "--package-path")
+            {
+                Description = "Path, underneath the 'sdk' folder (ex: devcenter/devbox)",
+                IsRequired = true,
+            };
+
+            engPathOption = new Option<string>(
+                "--eng-path")
+            {
+                Description = "Path to an 'eng' folder within an SDK repo (ex: /home/user/src/azure-sdk-for-go/eng)",
+                IsRequired = true,
+            };
+
+            outputPathOption = new Option<string>(
+                "--output-path")
+            {
+                Description = "Path to write the generated README contents",
+                IsRequired = true,
```

Comment:
Is this required or optional? Can this path be defaulted to package root path?

Reply:
I want to default this to something so badly, but there's no guarantee of what location they're in when they run this (or any) command.

I could try to default it to the current folder, and then fail if it turns out I can't derive a package path from it, however. What do you think of that?